server/service/microsoft_mdm.go

```diff
@@ -499,9 +499,9 @@ func validateBinarySecurityToken(ctx context.Context, encodedBinarySecToken stri
 			return fmt.Errorf("binarySecurityTokenValidation: host data cannot be found %v", err)
 		}
 
-		// Extra checks could be done here to ensure that host is ready for MDM enrollment
-		if host.Platform != "Windows" {
-			return fmt.Errorf("binarySecurityTokenValidation: host is not windows %v", err)
+		// This ensures  that only hosts that are elegible for Windows enrollment can be enrolled
+		if host.IsElegibleForWindowsMDMEnrollment() {
+			return fmt.Errorf("binarySecurityTokenValidation: host is not elegible for Windows MDM enrollment %v", err)
```

Comment:
The `err` will be `nil` here so there's nothing to wrap:

```suggestion
			return errors.New("binarySecurityTokenValidation: host is not elegible for Windows MDM enrollment")
```

Reply:
Oh and actually it should be `if !host.IsElegible...`.

Reply:
yes, and I also realized that I should add a test to test that host that is not eligible for Windows enrollment does not pass this check, so I've added a new test

Reply:
I've just fixed the error wrapping issue